models/fates/Dockerfile

```diff
@@ -0,0 +1,148 @@
+     make -j 2 install && \
+     ldconfig
+
+RUN  wget -q https://parallel-netcdf.github.io/Release/pnetcdf-1.12.1.tar.gz && \
+     tar zxf pnetcdf-1.12.1.tar.gz && \
+     cd pnetcdf-1.12.1 && \
+     ./configure MPIF77=/usr/local/bin/mpif77 MPIF90=/usr/local/bin/mpif90 MPICXX=/usr/local/bin/mpicxx MPICC=/usr/local/bin/mpicc && \
+     make -j 2 install && \
+     ldconfig
+
+RUN cd  /tmp/sources \
+    wget -q https://github.com/esmf-org/esmf/archive/refs/tags/ESMF_8_2_0.tar.gz && \
+    tar zxf ESMF_8_2_0.tar.gz && \
+    cd esmf-ESMF_8_2_0 && \
+    export ESMF_DIR='/tmp/sources/esmf-ESMF_8_2_0' && \
+    export ESMF_INSTALL_PREFIX='/usr/local' && \
+    export ESMF_LAPACK="netlib" && \
+    export ESMF_LAPACK_LIBS="-llapack -lblas" && \
+    export ESMF_LAPACK_LIBPATH="/usr/lib/x86_64-linux-gnu" && \
+    export ESMF_NETCDF="nc-config" && \
+    make && \
+    make install && \
+    ldconfig
+
+RUN rm -rf /tmp/sources
+RUN pip install --upgrade pip setuptools
+
+## Above has been tested in the pecan/model container
+
+########################################################################
+
+## Below needs further discussion.
+
+## Load CTSM from https://github.com/NorESMhub/noresm-land-sites-platform/blob/main/docker/api/entrypoint_api.sh
+    ## 1. Standard address of /ctsm-api/resources/model in pecan ??
+RUN if [ ! -d /ctsm-api/resources/model ]; then && \ 
+        mkdir -p /ctsm-api/resources && \
+    fi && \
+    ## go to the folder for CTSM
+    cd /ctsm-api/resources/ && \
+    ## download CTSM
+    git clone https://github.com/ESCOMP/CTSM.git && \
+    ## go to model folder
+    cd model && \
+    ## Checkout specific CTSM version, which has been tested in the container, more recent version than this needs to be tested in the future.
+    git checkout ctsm5.1.dev108 && \ 
+    ## Download fates (tag = sci.1.58.1_api.24.1.0), the version can be modified in Externals_CLM.cfg 
+    ./manage_externals/checkout_externals
+
+RUN set -e && \
+    ## set home folder for running CTSM-fates. 2. what does 'HOME' should be in pecan?
+    USER=root && \
+    HOME=/root && \
+    ## create .bashrc file for setting basic compiling parameters for CTSM-fates
+    export USER="$USER" && \
+    export PYTHONPATH=/ctsm-api && \
+    export CIME_MACHINE=container && \
+    export MPICC=mpicc && \
+    export MPIFC=mpif90 && \
+    export MPIF90=mpif90 && \
+    export MPIF77=mpif77 && \
+    ## 3. What is data folder for CTSM-fates in pecan?
+    export CESMDATAROOT=${CESMDATAROOT:?} && \
+    ln -fs /ctsm-api/docker/dotcime "$HOME"/.cime
+
+## Copy dotcime and entrypoints from ctsm-api to fates
+## Dotcime is a key file for setting the compiling environment for CTSM-fates 
+## Dotcime (https://github.com/NorESMhub/ctsm-api/tree/main/docker/dotcime) SHOULD be kept in the home directory as '.cime'
+RUN COPY docker/dotcime /<home folder>/.cime
+
+## 3. Download inputdata in dockerfile?
+## 4. Build CTSM-fates here or in write.config file?
```

Comment:
Do we need to build CTSM-fates in Dockerfile or in write.config.R?

Reply:
Since CLM-FATES needs to be re-complied every time when some setting changes, I would vote for putting the "building " phase of the model in [template.job](https://github.com/PecanProject/pecan/blob/develop/models/fates/inst/template.job) similar to what have been done there at line [134](https://github.com/PecanProject/pecan/blob/caad7b3f8386df43eaf44f9316459f66ffc69b0b/models/fates/inst/template.job#L134).